{"name": "FireBase_Verify Get Chatbot API when the env_id is invalid or missing -- @1.4 ", "status": "broken", "statusDetails": {"message": "requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD08yxn7V_YAfkAY8mKDTYjfKV_qoPrdRQ\n", "trace": "  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\model.py\", line 1991, in run\n    match.run(runner.context)\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\matchers.py\", line 105, in run\n    self.func(context, *args, **kwargs)\n  File \"features\\steps\\FireBase_Get_all_chatbots_for_a_given_environment.py\", line 23, in step_impl\n    id_token = get_firebase_id_token()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\utilities\\authentication.py\", line 16, in get_firebase_id_token\n    response.raise_for_status()  # Raise an error for non-2xx responses\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\requests\\models.py\", line 1026, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"}, "steps": [{"name": "Given FireBase_User sends GET request to fetch chatbots with env_id \"<script>alert('XSS')</script>\"", "status": "broken", "statusDetails": {"message": "requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD08yxn7V_YAfkAY8mKDTYjfKV_qoPrdRQ\n", "trace": "  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\model.py\", line 1991, in run\n    match.run(runner.context)\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\matchers.py\", line 105, in run\n    self.func(context, *args, **kwargs)\n  File \"features\\steps\\FireBase_Get_all_chatbots_for_a_given_environment.py\", line 23, in step_impl\n    id_token = get_firebase_id_token()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\utilities\\authentication.py\", line 16, in get_firebase_id_token\n    response.raise_for_status()  # Raise an error for non-2xx responses\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\requests\\models.py\", line 1026, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"}, "start": 1757691358372, "stop": 1757691359067}, {"name": "When FireBase_the user receives the response", "status": "skipped", "start": 1757691359071, "stop": 1757691359071}, {"name": "Then FireBase_error code should be 400 for bad request error", "status": "skipped", "start": 1757691359071, "stop": 1757691359071}, {"name": "And FireBase_response shows error message \"Bad Request\"", "status": "skipped", "start": 1757691359071, "stop": 1757691359071}, {"name": "And FireBase_the message should be \"Invalid input:\"", "status": "skipped", "start": 1757691359071, "stop": 1757691359071}], "parameters": [{"name": "env_id", "value": "<script>alert('XSS')</script>"}], "start": 1757691358371, "stop": 1757691359071, "uuid": "3ef09f23-9c48-4e96-bb26-014bb5ad1944", "historyId": "670d5a791f4c0313a98aedccbcfc77ae", "fullName": "FireBase_Verify Get Chatbot API: FireBase_Verify Get Chatbot API when the env_id is invalid or missing", "labels": [{"name": "severity", "value": "normal"}, {"name": "tag", "value": "Negative"}, {"name": "tag", "value": "Security"}, {"name": "tag", "value": "Test"}, {"name": "feature", "value": "FireBase_Verify Get Chatbot API"}, {"name": "framework", "value": "behave"}, {"name": "language", "value": "cpython3"}], "titlePath": ["features", "FireBase_Verify Get Chatbot API"]}